lib/doorkeeper/models/application_mixin.rb

```diff
@@ -19,31 +19,27 @@ module ApplicationMixin
 
     module ClassMethods
       def by_uid_and_secret(uid, secret)
-        where(uid: uid.to_s, secret: secret.to_s).limit(1).to_a.first
+        find_by(uid: uid.to_s, secret: secret.to_s)
       end
 
       def by_uid(uid)
-        where(uid: uid.to_s).limit(1).to_a.first
+        find_by(uid: uid.to_s)
       end
     end
 
     private
 
     def has_scopes?
       Doorkeeper.configuration.orm != :active_record ||
-        Application.new.attributes.include?("scopes")
+        Doorkeeper::Application.column_names.include?('scopes')
     end
 
     def generate_uid
-      if uid.blank?
-        self.uid = UniqueToken.generate
-      end
+      self.uid = UniqueToken.generate if uid.blank?
     end
 
     def generate_secret
-      if secret.blank?
-        self.secret = UniqueToken.generate
-      end
+      self.secret = UniqueToken.generate if secret.blank?
```

Comment:
> I think that code is readable in both cases, but with one line it is more elegant. thoughtbot/guides recommends to use multiline if to avoid ternary operators and nothing more.

The very first line in https://github.com/thoughtbot/guides/blob/master/style/ruby/README.md advices against "conditional modifiers". I can find the related discussion if you are interested in it.